src/commons/sagas/__tests__/BackendSaga.ts

```diff
@@ -262,16 +262,45 @@ describe('Test SUBMIT_ANSWER Action', () => {
   });
 
   test('when role is not student', () => {
-    const mockAnsweredAssessmentQuestion = { ...mockAssessmentQuestion, answer: '42' };
-    return expectSaga(BackendSaga)
-      .withState({ session: { role: Role.Staff } })
-      .call(showWarningMessage, 'Answer rejected - only students can submit answers.')
-      .not.call.fn(postAnswer)
-      .not.put.actionType(UPDATE_ASSESSMENT)
-      .not.put.actionType(UPDATE_HAS_UNSAVED_CHANGES)
-      .hasFinalState({ session: { role: Role.Staff } })
+    const mockAnsweredAssessmentQuestion: Question =
+      mockAssessmentQuestion.type === 'mcq'
+        ? { ...mockAssessmentQuestion, answer: 42 }
+        : { ...mockAssessmentQuestion, answer: '42' };
+    const mockNewQuestions: Question[] = mockAssessment.questions.slice().map(
```

Comment:
The `slice()` in this and related tests for `postAnswer` don't really seem necessary? Isn't any entry modified by `map` constructed as a new object by the spread operator (who wrote the original test even)?